src/ssmDocument/explorer/documentTypeNode.ts

```diff
@@ -0,0 +1,110 @@
+import { PlaceholderNode } from '../../shared/treeview/nodes/placeholderNode'
+import { makeChildrenNodes } from '../../shared/treeview/treeNodeUtilities'
+import { toArrayAsync, updateInPlace } from '../../shared/utilities/collectionUtils'
+import { DocumentItemNode } from './documentItemNode'
+
+export class DocumentTypeNode extends AWSTreeNodeBase {
+    private readonly documentNodes: Map<string, DocumentItemNode>
+
+    public constructor(
+        public readonly regionCode: string,
+        public documentType: string,
+        public readonly registryName: string
+    ) {
+        super('', vscode.TreeItemCollapsibleState.Collapsed)
+
+        this.setLabel()
+
+        this.documentNodes = new Map<string, DocumentItemNode>()
+    }
+
+    private setLabel() {
+        this.label = `${this.documentType}`
+    }
+
+    public update(documentType: string): void {
+        this.documentType = documentType
+        this.setLabel()
+    }
+
+    public async getChildren(): Promise<AWSTreeNodeBase[]> {
+        return await makeChildrenNodes({
+            getChildNodes: async () => {
+                await this.updateChildren()
+
+                return [...this.documentNodes.values()]
+            },
+            getErrorNode: async (error: Error) =>
+                new ErrorNode(
+                    this,
+                    error,
+                    localize('AWS.explorerNode.documentType.error', 'Error loading documentType ssm document items')
+                ),
+            getNoChildrenPlaceholderNode: async () =>
+                new PlaceholderNode(
+                    this,
+                    localize('AWS.explorerNode.documentType.noSsmDocument', `[No document found]`)
+                ),
+            sort: (nodeA: DocumentItemNode, nodeB: DocumentItemNode) =>
+                nodeA.documentName.localeCompare(nodeB.documentName),
+        })
+    }
+
+    public async updateChildren(): Promise<void> {
+        const client: SsmDocumentClient = ext.toolkitClientBuilder.createSsmClient(this.regionCode)
+        const documents = new Map<string, SSM.Types.DocumentIdentifier>()
+
+        let request: SSM.ListDocumentsRequest = {
+            Filters: [
+                {
+                    Key: 'DocumentType',
+                    Values: [this.documentType],
+                },
+            ],
+        }
+
+        if (this.registryName === 'Owned by me') {
+            request.Filters?.push({
+                Key: 'Owner',
+                Values: ['Self'],
+            })
+        } else if (this.registryName === 'Shared with me') {
+            request.Filters?.push({
+                Key: 'Owner',
+                Values: ['Private'],
+            })
+        } else if (this.registryName === 'Owned by Amazon') {
+            request.Filters?.push({
+                Key: 'Owner',
+                Values: ['Amazon'],
+            })
+        }
+
+        const docs = await toArrayAsync(client.listDocuments(request))
```

Comment:
This request + client query could be moved to a separate method to improve readability

Reply:
Will do